### PR TITLE
Add a unit test for python-file-like directory exclusion

### PR DESCRIPTION
**Pull request by harupy**
## Summary

Related to #5850

### CODE REVIEW

File: crates/ruff/src/resolver.rs [hunk 1]
```diff
@@ -606,4 +608,39 @@ mod tests {
...
+        let file1 = root.join("file1.py");
+        let dir1 = root.join("dir1.py");
+        let file2 = dir1.join("file2.py");
+        let dir2 = root.join("dir2.py");
+        // root
+        // ├── dir1.py
+        // │   └── file2.py
+        // ├── dir2.py
+        // └── file1.py
+        std::fs::File::create(&file1).unwrap();
```
Comment (konstin) [inline crates/ruff/src/resolver.rs:625]: nit: `use std::fs::File` and then `File::create(&file1).unwrap();`

File: crates/ruff/src/resolver.rs [hunk 2]
```diff
@@ -606,4 +608,39 @@ mod tests {
...
+            ),
+            &NoOpProcessor,
+        )
+        .unwrap()
+        .0;
+        let mut paths = paths.iter().flatten().collect::<Vec<_>>();
+        paths.sort_by(|a, b| a.path().cmp(b.path()));
+        assert_eq!(paths.len(), 2);
+        assert_eq!(paths[0].path(), file2);
+        assert_eq!(paths[1].path(), file1);
```
Comment (konstin) [inline crates/ruff/src/resolver.rs:644]:
```suggestion
        let mut paths = paths
            .iter()
            .flatten()
            .map(|entry| entry.path())
            .collect::<Vec<_>>();
        paths.sort();
        assert_eq!(paths, &[file2, file1]);
```

File: crates/ruff/src/resolver.rs [hunk 3]
```diff
@@ -606,4 +608,39 @@ mod tests {
...
+        let root = tmp_dir.path();
+        let file1 = root.join("file1.py");
+        let dir1 = root.join("dir1.py");
+        let file2 = dir1.join("file2.py");
+        let dir2 = root.join("dir2.py");
+        // root
+        // ├── dir1.py
+        // │   └── file2.py
+        // ├── dir2.py
+        // └── file1.py
```
Comment (konstin) [inline crates/ruff/src/resolver.rs:624]: good comment

**Review by konstin — APPROVED** [review]

**Comment by charliermarsh** [comment]
This is great! Thanks!

**Pull request merged**